spdlog/src/formatter/json_formatter.rs

```diff
@@ -0,0 +1,184 @@
+use std::fmt::{self, Write};
+
+use cfg_if::cfg_if;
+use serde::{ser::SerializeStruct, Serialize};
+
+use crate::{
+    formatter::{FmtExtraInfo, Formatter, LOCAL_TIME_CACHER},
+    Error, Record, StringBuf, __EOL,
+};
+
+struct JsonRecord<'a>(&'a Record<'a>);
+
+impl<'a> Serialize for JsonRecord<'a> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let src_loc = self.0.source_location();
+
+        let mut record =
+            serializer.serialize_struct("JsonRecord", if src_loc.is_none() { 4 } else { 5 })?;
+
+        record.serialize_field("level", &self.0.level())?;
+        {
+            let mut local_time_cacher = LOCAL_TIME_CACHER.lock();
+            record.serialize_field::<str>(
+                "time",
+                local_time_cacher
+                    .get(self.0.time())
+                    .full_second_str()
+                    .as_ref(),
+            )?;
+        }
+        record.serialize_field("tid", &self.0.tid())?;
+        record.serialize_field("payload", self.0.payload())?;
+
+        if let Some(src_loc) = src_loc {
+            record.serialize_field("source_location", src_loc)?;
```

Comment:
```suggestion
            record.serialize_field("source", src_loc)?;
```

Since we have already used name `{source}` in the pattern formatter, it's better to be consistent by using `source` as the field name here as well, it's also shorter.